src/lib.rs

```diff
@@ -1,21 +1,40 @@
 use http::Response;
 use hyper::body::{Body, Bytes, Frame, SizeHint};
-use pin_project::pin_project;
-use std::future::Future;
 use std::pin::Pin;
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::Arc;
 use std::task::{Context, Poll};
 use tonic::body::BoxBody;
-use tonic::metadata::MetadataValue;
 use tonic::transport::channel::ResponseFuture;
-use tonic::transport::Channel;
+use tonic::transport::{Channel, Error};
 use tonic::Status;
 use tower_service::Service;
 
+/// Struct for indicating if a frame has been produced.
```

Comment:
`Struct` is obvious, use something else... `Handle`? Or nothing, just `Indicates if a frame has been produced, and allows resetting the state.` or such